packages/fast-components-react-base/src/slider-track-item/slider-track-item.tsx

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { get } from "lodash-es";
+import { SliderTrackItemClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    SliderTrackItemAnchor,
+    SliderTrackItemHandledProps,
+    SliderTrackItemProps,
+    SliderTrackItemUnhandledProps,
+} from "./slider-track-item.props";
+import { SliderOrientation } from "../slider/slider.props";
+import { SliderContext, SliderContextType } from "../slider/slider-context";
+import { DisplayNamePrefix } from "../utilities";
+
+class SliderTrackItem extends Foundation<
+    SliderTrackItemHandledProps,
+    SliderTrackItemUnhandledProps,
+    {}
+> {
+    public static displayName: string = `${DisplayNamePrefix}SliderTrackItem`;
+
+    public static contextType: React.Context<SliderContextType> = SliderContext;
+
+    protected handledProps: HandledProps<SliderTrackItemHandledProps> = {
+        maxValuePositionBinding: void 0,
+        minValuePositionBinding: void 0,
+        managedClasses: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                style={{
+                    position: "absolute",
+                    ...this.applyPositioningValues(),
+                }}
+            >
+                {this.props.children}
+            </div>
+        );
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.sliderTrackItem", "");
+
+        if (
+            (this.context as SliderContextType).sliderOrientation ===
```

Comment:
Why do you need to type cast `this.context`?

Reply:
Apparently I don't...